app/src/main/java/ch/epfl/polychef/fragments/PostRecipeFragment.java

```diff
@@ -488,6 +499,57 @@ private void getAndCheckIngredients() {
         wrongInputs.put("Ingredients", true);
     }
 
+
+    private void initializeFromOriginalRecipe(Recipe originalRecipe) {
+        postingAModifiedRecipe=true;
+        this.originalRecipe=originalRecipe;
+
+        hideImageComponents();
+
+        EditText prepTimeInput = getView().findViewById(R.id.prepTimeInput);
+        prepTimeInput.setText(Integer.toString(originalRecipe.getEstimatedPreparationTime()));
+
+        EditText cookTimeInput = getView().findViewById(R.id.cookTimeInput);
+        cookTimeInput.setText(Integer.toString(originalRecipe.getEstimatedCookingTime()));
+
+        EditText personNb = getView().findViewById(R.id.personNbInput);
+        personNb.setText(Integer.toString(originalRecipe.getPersonNumber()));
+
+        EditText title=getView().findViewById(R.id.nameInput);
+        title.setText(originalRecipe.getName());
+
+        difficultyInput.setSelection(originalRecipe.getRecipeDifficulty().ordinal());
+
+        List<String> instructions=originalRecipe.getRecipeInstructions();
+        instructionText.setText(instructions.get(0));
+
+        for (int i=1;i<instructions.size();++i){
+            setAddInstructionButton(getView());
+            ((TextView)getView().findViewById(instructionsId.get(i))).setText(instructions.get(i));
+        }
+
+        List<Ingredient> ingredients=originalRecipe.getIngredients();
+
+        insertIngredientAtIndex(ingredients.get(0),0);
+        for (int i=1;i<ingredients.size();++i){
+            setAddIngredientButton(getView());
+            insertIngredientAtIndex(ingredients.get(i),i);
+        }
```

Comment:
Is it not possible to do everything in the loop?

Reply:
No because if there is only one ingredient we do not need to add a new field as there always exist one at the beginning
or I can write a if-condition inside the loop that only adds a button when i!=0

Reply:
Ah, makes sense! It's all good then